docs/01-start-here/06-testing-tips.md

```diff
@@ -0,0 +1,157 @@
+ that corresponds to the application in question and enter:
+```
+  test
+```
+To run all the scala unit tests. execute this command from the root project.
+
+To run only the tests for a particular test suite, use the command:
+```
+  test-only *<name-of-your-test-suite>
+```
+
+### How to test your devices on local
+- It is possible to use chrome emulator to get a good idea of how android devices behave and XCode the hardware simulators on XCode to do the same for iOS,
+However for significant frontend changes you should test on devices as much as possible. To do this you need to be able to make your localhost accessible.
+A tool called `ngrok` is a good way to make a local port accessible to external devices or computer (instructions on how to run ngrok [here](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/09-testing-externally-on-localhost.md))
+- To test on a variety of devices or to reproduce a bug on a particular set up, use https://www.browserstack.com. This site has a collection of computers and devices,
+and many different versions of browsers. It is a very useful testing resource. Contact the dotcom team to get added to our user account for this tool
+
+### What to test on
+A list of useful information on device testing follows
+[Our browser support principals](https://github.com/guardian/frontend/blob/master/docs/04-quality/02-browser-support-principles.md);
+[A list of what browsers we support](https://github.com/guardian/frontend/blob/master/docs/04-quality/01-browser-support.md);
+[Ranking of the current top browsers](https://github.com/guardian/frontend/blob/master/docs/04-quality/01-browser-support.md)
+[Breakdown of last seven days web traffic by type of device](https://dashboard.ophan.co.uk/graph/breakdown?days=7&device=Personal+computer&device=Smartphone&device=Tablet&by=device)
+
+### Good places/tricks to find bugs
+- iPhone - Safari is the most popular browser on iOS (which makes it one of the most popular ways to view our site) and even Chrome on iOS has to use the iOS Webkit framework.
+ It is interesting from a testing perspective because it is a different operating system, different browser setup and different screen-size from the machine you are developing on.
+ This combination often throws up interesting layout problems. In addition Safari often has less support for new features compared with Chome, which can also lead to rendering errors.
+
+- IE9 - This is the second most popular IE version. IE is notorious for having less support for features and also being much less tolerant of small errors than Chrome, or Firefox. Often when a page renders on those browsers it will still have problems on IE
+ Add to this the fact that it is much older than IE11 so supports even less features. IE9 is the oldest browser version that we expect to be fully supported. If it works on IE9 it is a good indication it will work on most other browsers.
```

Comment:
I don't think that's true anymore, last time we discussed I believe we agreed that IE11 was now the lowest we fully support. See https://frontend.gutools.co.uk/analytics/quality - all the older versions have now left our top browsers (even when cut my time of day).